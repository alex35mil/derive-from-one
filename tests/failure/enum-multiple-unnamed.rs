use derive_from_one::FromOne;

#[derive(FromOne)]
pub enum T {
    A(bool, u32),
    B(&'static str),
}

fn main() {
    let _: T = T::from("b");
    let _: T = T::from(true);
}
