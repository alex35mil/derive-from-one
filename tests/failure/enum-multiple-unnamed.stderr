error[E0277]: the trait bound `T: From<bool>` is not satisfied
  --> tests/failure/enum-multiple-unnamed.rs:11:24
   |
11 |     let _: T = T::from(true);
   |                ------- ^^^^ the trait `From<bool>` is not implemented for `T`
   |                |
   |                required by a bound introduced by this call
   |
   = help: the trait `From<&'static str>` is implemented for `T`
